docs/features/software-catalog/catalog-customization.md

```diff
@@ -5,64 +5,232 @@ title: Catalog Customization
 description: How to add custom filters or interface elements to the Backstage software catalog
 ---
 
-The Backstage software catalog comes with a default `CatalogIndexPage` to filter
-and find catalog entities. This is already set up by default by
-`@backstage/create-app`.
+The Backstage software catalog comes with a default `CatalogIndexPage` to filter and find catalog entities. This is already set up by default by `@backstage/create-app`. If you want to change the default index page - to set the initially selected filter, adjust columns, add actions, or to add a custom filter to the catalog - the following sections will show you how.
 
-If you want to change the default index page - such as to add a custom filter to
-the catalog - you can create your own `CatalogIndexPage`.
+## Pagination
 
-> Note: The catalog index page is designed to have a minimal code footprint to
-> support easy customization, but creating a copy does introduce a possibility
-> of drifting out of date over time. Be sure to check the catalog
-> [CHANGELOG](https://github.com/backstage/backstage/blob/master/plugins/catalog/CHANGELOG.md)
-> periodically.
+Initial support for pagination of the `CatalogIndexPage` was added in v1.21.0 of Backstage, so make sure you are on that version or newer to use this feature. To enable pagination you simply need to pass in the `pagination` prop like this:
 
-For example, suppose that I want to allow filtering by a custom annotation added
-to entities, `company.com/security-tier`. To start, I'll copy the code for the
-default catalog page and create a component.
+```tsx title="packages/app/src/App.tsx"
+<Route path="/catalog" element={<CatalogIndexPage pagination />} />
+```
 
-```tsx
-// imports, etc omitted for brevity. for full source see:
-// https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/CatalogPage/DefaultCatalogPage.tsx
-export const CustomCatalogPage = ({
```

Comment:
On this deletion, I'll say though, I think the original idea was that it opens up for really ANY type of customization fairly easily (at the cost of these lines of boilerplate). Like replacing the header and what's in it,  the support button ...

Might be worth retaining this as the advanced option, such that "if you need to go beyond these simpler options, here's how to replace the page as a whole using its constituent sub-parts" so to speak

Reply:
Let me see what I can do to add this back, it's a valid option for those who want to go much deeper customizing htis. The problem I was running into was that it wasn't as simple as copying what we as at - https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/CatalogPage/DefaultCatalogPage.tsx - I needed to pull in more "things".

Reply:
Alright, added back but had to do s bit of adjustments, think this worked enough to give people the right idea.